src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java

```diff
@@ -133,7 +133,7 @@ public boolean isBalanceDiffConsistent(Set<Hash> approvedHashes, Map<Hash, Long>
         }
         Set<Hash> visitedHashes = new HashSet<>(approvedHashes);
         Map<Hash, Long> currentState = generateBalanceDiff(visitedHashes, tip,
-                snapshotProvider.getLatestSnapshot().getIndex());
+                snapshotProvider.getLatestSnapshot().getIndex(), true);
```

Comment:
Forcing extra rules here cannot break backward compatibility?

Reply:
Please verify that when we apply milestone we don't hit this code
Only in tip-selection we do